syncstorage-db-common/src/lib.rs

```diff
@@ -68,190 +68,189 @@ impl<E> Clone for Box<dyn DbPool<Error = E>> {
 
-    fn get_bsos(&mut self, params: params::GetBsos) -> DbFuture<'_, results::GetBsos, Self::Error>;
+    async fn get_bsos(&mut self, params: params::GetBsos) -> Result<results::GetBsos, Self::Error>;
 
-    fn get_bso_ids(
+    async fn get_bso_ids(
         &mut self,
         params: params::GetBsos,
-    ) -> DbFuture<'_, results::GetBsoIds, Self::Error>;
+    ) -> Result<results::GetBsoIds, Self::Error>;
 
-    fn post_bsos(&mut self, params: params::PostBsos) -> DbFuture<'_, SyncTimestamp, Self::Error>;
+    async fn post_bsos(&mut self, params: params::PostBsos) -> Result<SyncTimestamp, Self::Error>;
 
-    fn delete_bso(
+    async fn delete_bso(
         &mut self,
         params: params::DeleteBso,
-    ) -> DbFuture<'_, results::DeleteBso, Self::Error>;
+    ) -> Result<results::DeleteBso, Self::Error>;
 
-    fn get_bso(
+    async fn get_bso(
         &mut self,
         params: params::GetBso,
-    ) -> DbFuture<'_, Option<results::GetBso>, Self::Error>;
+    ) -> Result<Option<results::GetBso>, Self::Error>;
 
-    fn get_bso_timestamp(
+    async fn get_bso_timestamp(
         &mut self,
         params: params::GetBsoTimestamp,
-    ) -> DbFuture<'_, results::GetBsoTimestamp, Self::Error>;
+    ) -> Result<results::GetBsoTimestamp, Self::Error>;
 
-    fn put_bso(&mut self, params: params::PutBso) -> DbFuture<'_, results::PutBso, Self::Error>;
+    async fn put_bso(&mut self, params: params::PutBso) -> Result<results::PutBso, Self::Error>;
 
-    fn create_batch(
+    async fn create_batch(
         &mut self,
         params: params::CreateBatch,
-    ) -> DbFuture<'_, results::CreateBatch, Self::Error>;
+    ) -> Result<results::CreateBatch, Self::Error>;
 
-    fn validate_batch(
+    async fn validate_batch(
         &mut self,
         params: params::ValidateBatch,
-    ) -> DbFuture<'_, results::ValidateBatch, Self::Error>;
+    ) -> Result<results::ValidateBatch, Self::Error>;
 
-    fn append_to_batch(
+    async fn append_to_batch(
         &mut self,
         params: params::AppendToBatch,
-    ) -> DbFuture<'_, results::AppendToBatch, Self::Error>;
+    ) -> Result<results::AppendToBatch, Self::Error>;
 
-    fn get_batch(
+    async fn get_batch(
         &mut self,
         params: params::GetBatch,
-    ) -> DbFuture<'_, Option<results::GetBatch>, Self::Error>;
+    ) -> Result<Option<results::GetBatch>, Self::Error>;
 
-    fn commit_batch(
+    async fn commit_batch(
         &mut self,
         params: params::CommitBatch,
-    ) -> DbFuture<'_, results::CommitBatch, Self::Error>;
+    ) -> Result<results::CommitBatch, Self::Error>;
 
-    fn check(&mut self) -> DbFuture<'_, results::Check, Self::Error>;
+    async fn check(&mut self) -> Result<results::Check, Self::Error>;
 
     fn get_connection_info(&self) -> results::ConnectionInfo;
 
     /// Retrieve the timestamp for an item/collection
     ///
     /// Modeled on the Python `get_resource_timestamp` function.
```

Comment:
We still need this here? If so, maybe a direct ref to the archived repo in question, otherwise maybe we don't need this comment.